### PR TITLE
Add picker for conflict calendars of appointments

### LINKED ISSUE #3722: Add calendar picker for appointment conflict selection
**Is your feature request related to a problem? Please describe.**

As a user I might have calendars that are full of events that do not reflect my actual availability. Then appointment slots are blocked by these events.

The `calendar_appt_configs.calendar_freebusy_uris` was added for this very purpose. We just have to add an UI and write the URIs to the DB.

**Describe the solution you'd like**

Allow the user to select which calendars should be used for the conflict check. As a default we'd still check all.

**Describe alternatives you've considered**

n/a

**Additional context**

n/a

### PR DESCRIPTION
**Pull request by st3iny**
Fix #3722 

Peek 2021-12-01 12-43

I refactored the existing `CalendarPicker` component to also work with multiple values so here is some proof that it still works elsewhere:

## Usages in editors

Screenshot 2021-12-01 at 12-44-58 Dezember 2021 - Kalender - Nextcloud
Screenshot 2021-12-01 at 12-45-13 Dezember 2021 - Kalender - Nextcloud

## Usages in read only shared editors

Screenshot 2021-12-01 at 12-32-54 Dezember 2021 - Kalender - Nextcloud
Screenshot 2021-12-01 at 12-40-40 Dezember 2021 - Kalender - Nextcloud

(Couldn't debug the last usage in the import modal because I failed to provide a valid ICS file.)

### CODE REVIEW

File: src/components/AppointmentConfigModal.vue [hunk 1]
```diff
@@ -206,6 +218,12 @@ export default {
 			const uri = this.editing.targetCalendarUri
 			return this.ownSortedCalendars.find(cal => this.calendarUrlToUri(cal.url) === uri)
 		},
+		conflictCalendars() {
+			const freebusyUris = this.editing.freebusyUris ?? []
+			return freebusyUris.map(uri => {
```
Comment (ChristophWurst) [inline src/components/AppointmentConfigModal.vue:223]: shouldn't it be `.filter`?

File: src/components/AppointmentConfigModal.vue [hunk 2]
```diff
@@ -206,6 +218,12 @@ export default {
 			const uri = this.editing.targetCalendarUri
 			return this.ownSortedCalendars.find(cal => this.calendarUrlToUri(cal.url) === uri)
 		},
+		conflictCalendars() {
+			const freebusyUris = this.editing.freebusyUris ?? []
```
Comment (ChristophWurst) [inline src/components/AppointmentConfigModal.vue:222]: unused?

Comment (st3iny) [reply src/components/AppointmentConfigModal.vue:222]: It's used in the next line.

File: src/components/AppointmentConfigModal.vue (hunk 1 above)
Comment (st3iny) [reply src/components/AppointmentConfigModal.vue:223]: We have to map it to calendar objects for the `CalendarPicker`. The config saves uri string and the picker accepts calendar objects.

Comment (st3iny) [reply src/components/AppointmentConfigModal.vue:223]: We have to map it to calendar objects for the `CalendarPicker`. The config saves uri string and the picker accepts calendar objects.

File: src/components/AppointmentConfigModal.vue (hunk 2 above)
Comment (st3iny) [reply src/components/AppointmentConfigModal.vue:222]: It's used in the next line.

Comment (ChristophWurst) [reply src/components/AppointmentConfigModal.vue:222]: right … got confused by the different colors here on the github diff …

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**